pkg/healthcheck/healthcheck.go

```diff
@@ -802,7 +802,14 @@ func getPodStatuses(pods []v1.Pod) map[string][]v1.ContainerStatus {
 	for _, pod := range pods {
 		if pod.Status.Phase == v1.PodRunning {
 			parts := strings.Split(pod.Name, "-")
-			name := strings.Join(parts[1:len(parts)-2], "-")
+			var name string
+			// All control plane pods  should have a name that results in 3 substrings string.Split on '-'
+			if len(parts) >= 3 {
+				name = strings.Join(parts[1:len(parts)-2], "-")
+			} else {
+				name = strings.Join(parts, "-")
```

Comment:
If we know that this isn't a control plane pod, we should just skip it altogether. I think this works:

```
			parts := strings.Split(pod.Name, "-")
			if len(parts) < 3 {
				// If this pod doesn't have a replicaset ID in it's name, then it's
				// not owned by the control plane and we can skip it
				continue
			}
			name := strings.Join(parts[1:len(parts)-2], "-")
```